ex9/lib/main.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+void main() => runApp(MaterialApp(home: FirestoreFirstDemo(),));
+
+FirestoreFirstDemoState pageState;
+
+class FirestoreFirstDemo extends StatefulWidget {
+  @override
+  FirestoreFirstDemoState createState() {
+    pageState = FirestoreFirstDemoState();
+    return pageState;
+  }
+}
+
+class FirestoreFirstDemoState extends State<FirestoreFirstDemo> {
+  final firestoreInstance =  Firestore .instance;
+  TextEditingController textEditingController = TextEditingController();
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text("Firestore Chat!")),
+      body: SingleChildScrollView(
+        physics: ScrollPhysics(),
+        scrollDirection: Axis.vertical,
+        child: Column(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: <Widget>[
+              Container(
+                height: 450,
```

Comment:
You shouldn't use hardcoded sizes because you don't know on what device the code will run. Can you wrap the ListView in an Expanded component?